src/schema/comments.ts

```diff
@@ -785,6 +810,32 @@ export const resolvers: IResolvers<any, Context> = {
       });
       return { _: true };
     },
+    reportComment: async (
+      source,
+      { commentId: id, reason, note }: ReportCommentArgs,
+      ctx: Context,
+    ): Promise<GQLEmptyResponse> => {
+      if (!reportCommentReasons.has(reason)) {
+        throw new ValidationError('Reason is invalid');
+      }
+
+      await ctx.getRepository(Comment).findOneByOrFail({ id });
```

Comment:
Do we need this, would `insert` not just fail due to referential integrity?

Reply:
While that can work, the `insert` is inside the try-catch and it ignores any thrown error.

Reply:
It would throw a TypeOrmError of foreign key constraint no?
I think you can indeed bypass this above line  (see for example saveHiddenPost function)

Reply:
I tried but the CommentReport schema seems to be missing the FK relating to the `commentId`. Do you want me to update the schema itself?

Reply:
I see post report also missing FK, it might be to avoid reports being deleted when post/comments are deleted?

Let's keep the query in my opinion then, don't wan to add additional work when minor thing.

Reply:
Ah, that makes sense, yes. To not cascade deleting the reports. Thank you both for the reviews! 🚀

Reply:
Yeah agreed, wasn't aware of the nuance of the missing FK, but let's leave it.
Not a big thing.

Reply:
@sshanzel can we please select the relevant fields only? In this case it's enough to ask for id. Comment can be big due to the content column

Reply:
Ah, that is true. I will keep that in check for future fetches.